tests/scripts/fetch-csi-operator-files.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+TARGET_DIR="deploy/examples/csi-operator"
+DIRECTORIES=("config/crd/bases" "config/rbac")
+RBAC_FILE_NAME="$TARGET_DIR/rbac.yaml"
+CRD_FILE_NAME="$TARGET_DIR/crds.yaml"
+
+mkdir -p $TARGET_DIR
+: >$RBAC_FILE_NAME
+: >$CRD_FILE_NAME
+
+function fetch_files() {
+    local DIRS=("$@")
+
+    for DIR in "${DIRS[@]}"; do
+        API_URL="https://api.github.com/repos/ceph/ceph-csi-operator/contents/$DIR"
```

Comment:
It seems like it might be less code to maintain to just do `git clone https://github.com/ceph/ceph-csi-operator` and then copy the files we want. WDYT?

Reply:
I can try for now, will see if this cause problem in CI I guess `git status` will get confused with clone if we add the new csi-operator CI

Reply:
If this isn't critical, could we take note and address later?

Reply:
From huddle, it seems like there might have been some confusion around my suggestion. I don't think we want to add `ceph-csi-operator` as a "submodule", or to keep the cloned dir permanently. That would be too much I think. Just cloning into a temp location with the script and then copying the files we want sounds simple and doesn't rely on github's file API being stable long-term. I don't know what stability guarantees there will or won't be there, but I know `git` will be very stable.

So this might look something like this:

```sh
tmpdir="$(mktemp)"
( cd "$tmpdir"
  git clone https://github.com/ceph/ceph-csi-operator
)

cp "$tmpdir"/ceph-csi-operator/<path-to-file> <rook location>
# etc....

rm -rf "$tmpdir" # cleanup
```

Reply:
Anyway there will be improvements later to this script when we move to stable mode.

Reply:
@BlaineEXE I'm starting to do the change but I feel `git clone` may get a little messy in local deployment as it will always do the clone. We can always remove this but thinking is this a good approach? I'm happy to make this change but I wanted to raise this. let me know.

Reply:
> I feel git clone may get a little messy in local deployment as it will always do the clone.

That's why I suggested cloning into the temp dir created by `mktemp` (which usually goes into `/tmp`) and then doing `rm -rf <tmpdir>` after.

Since you and Travis have reservations, @subhamkrai I think you should make the decision. If the script breaks or requires updates, we will rely on you to fix/adjust it, so you may as well take ownership of the implementation. I don't want to make "this is what I would do" to be the same as "this is what you must do".

The solution implemented here has a possibility of being faster than a git clone because it pulls fewer files overall. Slow tooling is annoying, so this could be a good priority and implementation. Though as a comparative, git clone can also pull the files in parallel and without as many CLI commands, so I'm not sure which is faster overall, or by how much.

For tooling that I maintain, I want to touch the tooling as little as possible, and when I do have to, I want to be able to re-learn what the tooling does so I can modify it quickly. I find clone-and-cp to be conceptually simpler to grok, and my intuition says it's less likely to break unexpectedly, so that's what I would do.

If you find the clone-and-cp implementation to be confusing or think it'll have limitations, that will make it hard for you to maintain, which is important.

Reply:
Let me ask a higher level question... Should this script even be pulling from some URL or cloning? Actually I don't think so. This is a developer helper script, not a CI script. The developer should own cloning the csi-operator repo and ensure it is sync'd to whatever version of the CSI operator that is required. Sometimes we may want to sync a specific branch or tag of the CSI operator instead of the latest, in which case the dev can do that before running this script. All the script needs to do is look in the local directory structure under `$GOPATH/src/github.com/ceph/ceph-csi-operator` and copies the files directly from there.

Reply:
After running the script, the dev will open a PR and be reviewed before merge. So if the wrong version of the files is pulled, hopefully the process will catch that. I'm also concerned about the scenario where we need to pull a specific version or branch instead of just pulling main. Maybe that would be an argument to the script. 

But for now it's not a priority, it's fine for now, we just need to focus on critical items to get experimental mode ready for 1.15 and this is not critical.